bin/lib/notify.py

```diff
@@ -0,0 +1,134 @@
+        f"https://api.github.com/{path}",
+        None,
+        {
+            "User-Agent": USER_AGENT,
+            "Authorization": f"token {token}",
+            "Accept": "application/vnd.github.v3+json",
+        },
+    )
+    result = urllib.request.urlopen(req)
+    # It's ok not to check for error codes here. We'll throw either way
+    return json.loads(result.read())
+
+
+def paginated_get(entity: str, token: str, query: dict = None) -> List[dict]:
+    if query is None:
+        query = {}
+    result: List[dict] = []
+    results_per_page = 50
+    query["page"] = 1
+    query["per_page"] = results_per_page
+    while True:
+        current_page_results = get(entity, token, query)
+        result.extend(current_page_results)
+        if len(current_page_results) == results_per_page:
+            query["page"] += 1
+        else:
+            break
+    return result
+
+
+def list_inbetween_commits(end_commit: str, new_commit: str, token: str) -> List[dict]:
+    commits = get(f"repos/{OWNER_REPO}/compare/{end_commit}...{new_commit}", token=token)
+    return commits["commits"]
+
+
+def get_linked_pr(commit: str, token: str) -> dict:
+    pr = get(f"repos/{OWNER_REPO}/commits/{commit}/pulls", token=token)
+    return pr
+
+
+def get_linked_issues(pr: str, token: str):
+    query = """
+    query {
+        repository(owner: "compiler-explorer", name: "compiler-explorer") {
+            pullRequest(number: %s) {
+              closingIssuesReferences(first: 10) {
+                edges {
+                  node {
+                    number
+                  }
+                }
+              }
+            }
+        }
+    }
+    """ % pr
+    return post("graphql", token, {"query": query})
+
+
+def get_issue_comments(issue: str, token: str) -> List[dict]:
+    return paginated_get(f"repos/{OWNER_REPO}/issues/{issue}/comments", token)
+
+
+def comment_on_issue(issue: str, msg: str, token: str):
+    result = post(f"repos/{OWNER_REPO}/issues/{issue}/comments", token, {"body": msg}, dry=True)
+    return result
+
+
+def send_live_message(issue: str, token: str):
+    comment_on_issue(issue, "This is now live", token=token)
+
+
+def get_edges(issue: dict) -> List[dict]:
+    return issue["data"]["repository"]["pullRequest"]["closingIssuesReferences"]["edges"]
+
+
+def handle_notify(base, new, token):
+    print(f'Notifying from {base} to {new}')
+
+    commits = list_inbetween_commits(base, new, token)
+
+    prs = [get_linked_pr(commit["sha"], token) for commit in commits]
+    ids = [pr[0]["number"] for pr in prs]
+    linked_issues = [get_linked_issues(pr, token) for pr in ids]
+    issues_ids = [get_edges(issue) for issue in linked_issues if len(get_edges(issue)) > 0]
+
+    for edge in issues_ids:
+        for node in edge:
+            issue = node["node"]["number"]
+            comments = get_issue_comments(issue, token)
+            if not any(["This is now live" in comment["body"] for comment in comments]):
```

Comment:
As above, this might be easier on humans (and even the GraphQL above) if it used a label, but this also works (although relies on the exact message, which should probably be a `CONSTANT` if we go this way).

Reply:
There does not seem to be a way to filter by labels, but we can ask for them and do the filtering ourselves. In fact, we could even skip the rest api althogether and just query graphql with
```graphql
{
  repository(owner: "compiler-explorer", name: "compiler-explorer") {
    defaultBranchRef {
      target {
        ... on Commit {
          history(first: 15, before: "646e641b144ea9c0aaa1fa54d1353ef7ff00ae79 0") {
            edges {
              node {
                ... on Commit {
                  associatedPullRequests(first: 10) {
                    edges {
                      node {
                        number
                        labels(first: 10) {
                          edges {
                            node {
                              name
                            }
                          }
                        }
                        closingIssuesReferences(first: 10) {
                          edges {
                            node {
                              number
                              labels(first: 10) {
                                edges {
                                  node {
                                    name
                                  }
                                }
                              }
                            }
                          }
                        }
                      }
                    }
                  }
                }
              }
            }
          }
        }
      }
    }
  }
}
```
The only issue with this would be pagination of the commits (As I have no idea how to properly achieve that currently)

Reply:
got it...I am happy to defer to you on this :)